fbrp/src/facebook_robotics_platform/setup/runtime/base.py

```diff
@@ -11,22 +15,49 @@ def __init__(self):
     async def run(self, name: str, proc_def: ProcDef, args: argparse.Namespace):
         raise NotImplementedError("Launcher hasn't implemented run!")
 
+    def get_pid(self):
+        raise NotImplementedError("Launcher hasn't implemented get_pid!")
+
+    def get_down_event(self):
+        if not hasattr(self, "_down_requested_event"):
+            self._down_requested_event = asyncio.Event()
+        return self._down_requested_event
```

Comment:
TIL about the `hasattr` check for introducing new attrs. Thx :) 

So what happens  to `self._down_requested_event` once we are done with down ? Should it be cleared ? Or is the assumption that once we issue a `down` the runtime is not usable any more and only logical thing to do after that is to destroy it ?